File: frontend/src/lib/api/mosqueSettings.ts
import apiClient from './client';

export interface MosqueSettings {
    id: string;
    latitude: number;
    longitude: number;
    address: string;
    calculationMethod: number;
    prayersJson: string;
    isLiveAnnouncementActive?: boolean;
    liveAnnouncementText?: string;
    liveAnnouncementStartDate?: string;
    liveAnnouncementEndDate?: string;
}

export interface UpdateMosqueSettingsDto {
    latitude: number;
    longitude: number;
    address: string;
    calculationMethod: number;
    prayersJson: string;
    isLiveAnnouncementActive?: boolean;
    liveAnnouncementText?: string;
    liveAnnouncementStartDate?: string;
    liveAnnouncementEndDate?: string;
}

export const fetchMosqueSettings = async (): Promise<MosqueSettings> => {
    const response = await apiClient.get<MosqueSettings>('/api/MosqueSettings');
    return response.data;
};

export const saveMosqueSettings = async (settings: UpdateMosqueSettingsDto): Promise<MosqueSettings> => {
    const response = await apiClient.post<MosqueSettings>('/api/MosqueSettings', settings);
    return response.data;
};

File: backend/src/QuranSchool.Domain/Entities/MosqueSettings.cs
using QuranSchool.Domain.Common;

namespace QuranSchool.Domain.Entities;

public class MosqueSettings : BaseEntity
{
    public double Latitude { get; set; } = 45.5019; // Default to existing value
    public double Longitude { get; set; } = -73.5674;
    public string Address { get; set; } = string.Empty;
    public int CalculationMethod { get; set; } = 2; // ISNA by default
    public string PrayersJson { get; set; } = string.Empty; // Stored JSON array of prayers and offsets
    
    // Nouveaux champs pour le Bandeau Annonce Live dynamique
    public bool IsLiveAnnouncementActive { get; set; } = false;
    public string? LiveAnnouncementText { get; set; }
    public DateTime? LiveAnnouncementStartDate { get; set; }
    public DateTime? LiveAnnouncementEndDate { get; set; }

    // BaseEntity already contains SchoolId
    public virtual School? School { get; set; }
}
